### ISSUE TITLE
Cannot set window icon: "Image load error: IO error"

### ISSUE THREAD

**Issue by Candunc** [ggez/ggez#554]
My context builder is below, where all constants are &'static str. ICON is a valid path, as I can call Image::new and graphics::draw, however I can't set it as an image icon, and the error isn't terribly useful.
```
const ICON: &'static str = "/icon.png";
[...]
let cb = ggez::ContextBuilder::new(GAME_ID, AUTHOR)
    .window_setup(WindowSetup::default().title(GAME_ID).icon(ICON));
```
```
    Finished dev [optimized + debuginfo] target(s) in 6.49s
     Running `target\debug\imperialism.exe`
Error: ResourceLoadError("Image load error: IO error")
error: process didn't exit successfully: `target\debug\mygame.exe` (exit code: 1)
```

My files are located at './target/debug/resources/', and I'm building on Windows 10 with the latest stable version of Rust. I tried some different filetypes as well, such as png and ico.

**Comment by icefoxen** [comment]
Responsible code: `src/graphics/context.rs` line 125 .  It looks like it uses winit's `Icon::from_path()` to load the icon instead of attempting to go through the `filesystem` module and its resource handling.  It should load the icon data through `filesystem`, one way or another, and pass it to `winit::Icon::from_rgba()`.  To confirm that this is the problem, try changing `ICON` to `./resources/icon.png` or wherever your icon is relative to your executable and/or wherever you run `cargo` from and see if that does it.

Oops.  Good catch!  Should be a fairly easy fix, if that is indeed the issue.  While we're at it we might want to add an option to load the icon from a `&[u8]` directly anyway, it might be convenient to be able to `include_bytes!()` your icon file.

**Comment by Candunc** [comment]
Changing the path to ```./resources/mygame.png``` works! I'll see if I can submit a pull request so that it depends on ```filesystem``` like I expected.

**Comment by icefoxen** [comment]
Sure, if you want.  It might be easiest to actually load the icon file in `Context::from_conf()` and then pass an `Option<Vec<u8>>` or whatever to `GraphicsContext::new()`.

**Comment by Candunc** [comment]
Closed in #555
